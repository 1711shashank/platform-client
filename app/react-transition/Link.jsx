import React from "react";
import PropTypes from "prop-types";

class Link extends React.Component {
  static getDerivedStateFromProps(nextProps, prevState) {
    const injector = angular.element(document).injector();
    const state = injector.get("$state");
    return {
      stateProvider: state
    };
    // not sure if this is working here
    if (nextProps.to !== prevState.to) {
      return {
        to: this.constructedTo(nextProps.to)
      };
    }
  }

  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
    this.state = {
      stateProvider: undefined,
      to: this.constructedTo(this.props.to)
    };
  }

  constructedTo(toFromProps) {
    // Strings need be defined before concatenating
    const to =
      typeof toFromProps === "string"
        ? toFromProps
        : toFromProps.pathname + toFromProps.search + toFromProps.hash;
    return to;
  }

  handleClick(event) {
    event.preventDefault();
    this.props.replace
      ? this.state.stateProvider.go(
          this.state.to,
          {},
          { location: this.props.replace }
        )
      : this.state.stateProvider.go(this.state.to);
  }

  render() {
    return (
      <div>
        <a
          href="#"
          onClick={this.handleClick}
          title={this.props.title}
          id={this.props.id}
          className={this.props.className}
        >
          {this.props.children}
        </a>
      </div>
    );
  }
}

Link.defaultProps = {
  replace: false,
  title: '',
  id: '',
  className: ''
};

Link.propTypes = {
  to: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.objectOf(PropTypes.string)
  ]).isRequired,
  replace: PropTypes.bool,
  title: PropTypes.string,
  id: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default Link;
