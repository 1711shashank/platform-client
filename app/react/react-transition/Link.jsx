import React from "react";
import PropTypes from "prop-types";
import angular from "angular";

// Note: Link objects and innerRef are not implemented.
// If you need them, you'll have to implement them yourself
// Talk to Carolyn about starter code

class Link extends React.Component {
    static getDerivedStateFromProps() {
        const injector = angular.element(document).injector();
        const state = injector.get("$state");
        return {
            stateProvider: state
        };
    }

    constructor(props) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
        this.state = {
            stateProvider: undefined
        };
    }

    handleClick() {
        console.log(this.props);
        return this.props.replace
            ? this.state.stateProvider.go(
                  this.props.to,
                  {},
                  { location: this.props.replace }
              )
            : this.state.stateProvider.go(this.props.to);
    }
/* eslint-disable */
  render() {
    return (
      <div>
        <a
          onClick={this.handleClick}
          href="#"
          title={this.props.title}
          id={this.props.id}
          className={this.props.className}
        >
          {this.props.children}
        </a>
      </div>
    );
  }
}
/* eslint-enable */

Link.defaultProps = {
    replace: false,
    title: "",
    id: "",
    className: ""
};

Link.propTypes = {
    to: PropTypes.string.isRequired,
    replace: PropTypes.bool,
    title: PropTypes.string,
    id: PropTypes.string,
    className: PropTypes.string,
    children: PropTypes.node.isRequired
};

export default Link;
