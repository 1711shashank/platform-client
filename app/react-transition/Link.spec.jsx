import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Link from "./Link";

test("Link accepts a route string, replace bool, id, className, title, and children", () => {
  const component = renderer.create(
    <Link to="posts.data" replace id="test" className="test" title="test">
      Click me!
    </Link>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Link routes to new component", () => {
  const mockNavigation = jest.fn();
  const link = shallow(<Link to="posts.data">Click me!</Link>);
  link.find("a").simulate("click");
  expect(mockNavigation.mock.calls.length).toEqual(1);
});
