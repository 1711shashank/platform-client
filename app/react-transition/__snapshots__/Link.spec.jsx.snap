// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link accepts a route string, replace bool, id, className, title, and children 1`] = `
<div>
  <a
    className="test"
    href="#"
    id="test"
    onClick={[Function]}
    title="test"
  >
    Click me!
  </a>
</div>
`;
